pythonturtle/turtleprocess.py

```diff
@@ -51,11 +53,44 @@ def send_report(self):
 
     def run(self):
 
-        builtins.help = builtins.license = builtins.exit = \
-                                lambda *args, **kwargs: print('Not supported')
-
         self.turtle = Turtle()
 
+        def exit():
+            """
+            The built-in exit function terminates the interpreter hence
+            making the application unusable. We terminate the
+            application directly instead. NOTE: We use a proxy function
+            to hide the implementation details from the user.
+            """
+            evt = wx.CommandEvent(wx.ID_EXIT, wx.ID_EXIT)
+            wx.PostEvent(self.window, evt)
+
+        builtins.exit = exit
+
+        def help(object=None):
+            """
+            The built-in help function prints to the console and is
+            interactive there, thus blocking the UI. We simply display
+            the docstring of an object, when called with an argument,
+            or show the application help screen otherwise.
+            """
+            if object:
+                print(object.__doc__)
+            else:
+                self.window.show_help()
```

Comment:
This doesn't work either. For some reason the window has lost some attributes:

```python
>>> help()
Traceback (most recent call last):
  File "/usr/lib/python3.6/code.py", line 91, in runcode
    exec(code, self.locals)
  File "<console>", line 1, in <module>
  File "/home/peter/PythonTurtle/pythonturtle/turtleprocess.py", line 80, in help
    self.window.show_help()
  File "/home/peter/PythonTurtle/pythonturtle/application.py", line 177, in show_help
    self.help_menu_item.Check()
AttributeError: 'ApplicationWindow' object has no attribute 'help_menu_item'
```

@cool-RR, do you know a way to trigger the "Teach me" button or the \<F1\> key?

Reply:
Did you try calling `ApplicationWindow.show_help`?

Reply:
That's what the code does, isn't it?

Reply:
Oops, I'm a dumbass. I could investigate, but this is very unimportant. You can make it a no-op in the case where an object wasn't specified.